Monika After Story/game/script-topics.rpy

```diff
@@ -6937,6 +6937,307 @@ label monika_citizenship:
+
+label monika_grad_speech_call:
+    if not renpy.seen_label("monika_grad_speech"):
+        m 2eub "Of course, [player]. I'd love to give you my graduation speech now!"
+        m 2eka "I just want to make sure that you have enough time to hear it, though. Remember, it takes about four minutes."
+
+        #making sure player has time
+        menu:
+            "I have time.":
+                m 4hub "Great!"
+                m 4eka "I hope you like it! I worked really, {i}really{/i} hard on it."
+                
+                #say speech
+                call monika_grad_speech
+
+                #timed menu to see if player listened
+                show screen mas_background_timed_jump(10, "monika_grad_speech_not_paying_attention")
+                menu:
+                    m "Well [player]? What do you think?"
+
+                    "It's great! I'm so proud of you!":
+                        hide screen mas_background_timed_jump
+                        $ mas_gainAffection(amount=5, bypass=True)
+                        m 2subfb "Aww, [player]!"
+                        m 2ekbfa "Thank you so much! I worked really hard on that speech, and it means so much that you're proud of me~"
+                        show monika 5eubfu at t11 zorder MAS_MONIKA_Z with dissolve
+                        m 5eubfu "As much as I wish I could have given my speech in front of everyone, just having you by my side is so much better."
+                        m 5eubfb "I love you so much, [player]!"
+
+                    "I like it!":
+                        hide screen mas_background_timed_jump
+                        $mas_gainAffection(amount=3, bypass=True)
+                        m 2eua "Thanks [player]!"
+                        m 4hub "I'm glad you enjoyed it!"
+
+                    "That {i}was{/i} long":
+                        hide screen mas_background_timed_jump
+                        $mas_loseAffection()
+                        m 2tkc "Well, I {i}did{/i} warn you, didn't I?"
+                        m 2dfc "..."
+                        m 2tfc "I spent {i}so{/i} much time on it and that's all you have to say?"
+                        m 6lktdc "I really thought after I told you how important this was to me, you would have been more supportive and let me have my moment."
+                        m 6ektdc "All I wanted was for you to be proud of me, [player]."
+                    
+                return
+
+            "I don't.":
+                m 2eka "Don't worry, [player]. I'll give my speech whenever you want~"
+                return
+                
+    #if you want to hear it again
+    else:
+        #did you timeout once?
+        if not renpy.seen_label("monika_grad_speech_not_paying_attention") or persistent._mas_pm_monika_listened is True:
```

Comment:
dont need to check for is True since this var can only be True/False/None and both None and False are equivalent to not True.
```suggestion
        if not renpy.seen_label("monika_grad_speech_not_paying_attention") or persistent._mas_pm_monika_listened:
```